source/files/dico/dictionnary.c

```diff
@@ -15,6 +15,7 @@ Dico* htreetodico(Tree* huff, Tree* ascii)
         (tree)->value = get_bin(ascii->data, huff, 0);
         (tree)->left = htreetodico(huff, ascii->left);
         (tree)->right = htreetodico(huff, ascii->right);
+        free(ascii);
```

Comment:
if you free ASCII here then you have to delete line 130 of `compression.c` where the dictionary is free. Moreover if you free like that you don't free the value which is dynamically instantiated.

`compression.c`
```c
    // free variables
    free(transcription);
    d_free(dictionary);
```

Reply:
I free "occurrences" with the free ascii here. I'm not freeing tree, you're tired X) I have put a comment near to the concerned line (95 in compression.c)
ascii is a tree with a data (char), occurence (int) and two child (Tree*) which are free before because of the recurtion. No more things to free

Reply:
indeed what I said is false and i'm also tired. You should prefer to free _ascii_ in the `compress` function to keep coherence but that's not mondatory.

Reply:
It would keep coherence, but we will lose 0.000001 second for going threw it a new time Kappa